src/cmap/connection.ts

```diff
@@ -369,7 +372,29 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
 
     // always emit the message, in case we are streaming
     this.emit('message', message);
-    const operationDescription = this[kQueue].get(message.responseTo);
+    let operationDescription = this[kQueue].get(message.responseTo);
+
+    // Protect against multiplexing.
+    if (this[kQueue].size > 1) {
```

Comment:
we'd want to add a test for a non-monitoring connection for this

Reply:
Added.